SDL/Constants.cuh

```diff
@@ -1,78 +1,124 @@
+using Vec1d = alpaka::Vec<Dim1d,Idx>;
+using QueueProperty = alpaka::NonBlocking;
+using WorkDiv = alpaka::WorkDivMembers<Dim, Idx>;
+
+Vec const elementsPerThread(Vec::all(static_cast<Idx>(1)));
+
+// - AccGpuCudaRt
+// - AccCpuThreads
+// - AccCpuFibers
+// - AccCpuSerial
+#ifdef ALPAKA_ACC_GPU_CUDA_ENABLED
+    using Acc = alpaka::AccGpuCudaRt<Dim, Idx>;
+#elif ALPAKA_ACC_CPU_B_SEQ_T_THREADS_ENABLED
+    using Acc = alpaka::AccCpuThreads<Dim, Idx>;
+#elif ALPAKA_ACC_CPU_B_SEQ_T_FIBERS_ENABLED
+    using Acc = alpaka::AccCpuFibers<Dim, Idx>;
+#elif ALPAKA_ACC_CPU_B_SEQ_T_SEQ_ENABLED
+    using Acc = alpaka::AccCpuSerial<Dim, Idx>;
 #endif
-#ifdef FP16_seg // changes segment values
-#define __F2H_seg __float2half  
-#define __H2F_seg __half2float  
-typedef __half FPX_seg;
-#else
-#define __F2H_seg
-#define __H2F_seg
-typedef float FPX_seg; 
+
+// Needed for files that are compiled by g++ to not throw an error.
+// uint4 is defined only for CUDA, so we will have to revisit this soon when running on other backends.
+#ifndef ALPAKA_ACC_GPU_CUDA_ENABLED
+struct uint4
+{
+    unsigned int x;
+    unsigned int y;
+    unsigned int z;
+    unsigned int w;
+};
 #endif
 
-#ifdef __CUDACC__
-#define CUDA_CONST_VAR __device__
+auto const devHost = alpaka::getDevByIdx<alpaka::DevCpu>(0u);
+auto const devAcc = alpaka::getDevByIdx<Acc>(0u);
+using QueueAcc = alpaka::Queue<Acc, QueueProperty>;
+
+// Buffer type for allocations where auto type can't be used.
+template<typename TAcc, typename TData>
+using Buf = alpaka::Buf<TAcc, TData, Dim1d, Idx>;
+
+template<typename T, typename TAcc, typename TSize, typename TQueue>
+ALPAKA_FN_HOST ALPAKA_FN_INLINE Buf<TAcc, T> allocBufWrapper(TAcc const & devAccIn, TSize nElements, TQueue queue) {
+#ifdef CACHE_ALLOC
+    return cms::alpakatools::allocCachedBuf<T, Idx>(devAccIn, queue, Vec1d(static_cast<Idx>(nElements)));
 #else
-#define CUDA_HOSTDEV
-#define CUDA_CONST_VAR
-#define CUDA_DEV
+    return alpaka::allocBuf<T, Idx>(devAccIn, Vec1d(static_cast<Idx>(nElements)));
 #endif
+}
+
+template<typename T, typename TAcc, typename TSize>
+ALPAKA_FN_HOST ALPAKA_FN_INLINE Buf<TAcc, T> allocBufWrapper(TAcc const & devAccIn, TSize nElements) {
+    return alpaka::allocBuf<T, Idx>(devAccIn, Vec1d(static_cast<Idx>(nElements)));
+}
+
+
+const unsigned int MAX_BLOCKS = 80;
+const unsigned int MAX_CONNECTED_MODULES = 40;
+const unsigned int N_MAX_PIXEL_MD_PER_MODULES = 100000;
+const unsigned int N_MAX_PIXEL_SEGMENTS_PER_MODULE = 50000;
+
+const unsigned int N_MAX_PIXEL_TRIPLETS = 5000;
+const unsigned int N_MAX_PIXEL_QUINTUPLETS = 15000;
+
+const unsigned int N_MAX_TRACK_CANDIDATES = 1000;
+const unsigned int N_MAX_PIXEL_TRACK_CANDIDATES = 4000;
+
+const unsigned int N_MAX_TRACK_CANDIDATE_EXTENSIONS = 200000;
+const unsigned int N_MAX_TRACK_EXTENSIONS_PER_TC = 30;
+const unsigned int N_MAX_T3T3_TRACK_EXTENSIONS = 40000;
+
+const unsigned int size_superbins = 45000;
+
+// Temporary fix for endcap buffer allocation.
```

Comment:
To remove these variables from the global scope in a later PR hopefully. And to do it in a way that also works for the cmssw integration. That reminds me though, let me put some relevant checks in the code for now so that this doesn't cause problems in the meantime.

Reply:
In the past, printout statements were causing an increase in the number of registers for some kernels, most notoriously for the T5 one. Can you check that this is not the case here?

In any case however, I would wrap those within the `Warnings` `ifdef` and maybe agree with people to run with that enabled when checking PRs? I would go ahead with wrapping the printouts no matter what and we can discuss next week on the validation procedure. I trust that these checks don't output anything for now, right?

Reply:
@VourMa These checks only occur on the CPU, so their cost should only be checking one condition each (the one condition inside the if statement). All of this is outside of the main timers as well (map loading, etc.) so I don't think this is an issue. My concern with wrapping these in the Warnings ifdef is that if these were to fail, it's possible they could fail silently and cause issues.

Reply:
I agree that this could potentially be an issue and that's why I proposed to define a workflow for checking it. However, I get your point about these being only on the CPU (sorry, I had not noticed), so they will most likely cause no issue. Let's not do anything about them for now but keep an eye for increased registers in the profiling report you are going to produce, just to make sure that there isn't anything we have overlooked.